### PR TITLE
Workflow and remote-user related refactoring and a bug fix

### PR DESCRIPTION
**Pull request by dannon**
xref #6075 and #5955, this is stuff I cleaned up while digging around.  I'm going to follow up on another remote user issue where there was sometimes no history set when there probably should be, but that'll happen separately.

### CODE REVIEW

**Comment by mvdbeek** [comment]
If you're already in the workflow editor, could you take a look at https://github.com/galaxyproject/galaxy/issues/5879 ? That's a nasty bug as you can get unexpected results.

File: lib/galaxy/web/framework/webapp.py [hunk 1]
```diff
@@ -643,6 +643,45 @@ def user_checks(self, user):
         """
         self.check_user_library_import_dir(user)
 
+    def associate_user_history(self, user, prev_galaxy_session=None):
+        # Associated the current user's last accessed history (if exists) with their new session
```
Comment (nsoranzo) [inline lib/galaxy/web/framework/webapp.py]: s/Associated/Associate/

Also, you can use the `"""`...`"""` syntax for proper method documentation.

**Comment by mvdbeek** [comment]
I think I got the fix for https://github.com/galaxyproject/galaxy/issues/5879, will open a separate PR

File: lib/galaxy/web/framework/webapp.py (hunk 1 above)
Comment (dannon) [reply lib/galaxy/web/framework/webapp.py]: Fixed nine other typos while browsing through, but missed the one here in the comment text I actually copied, of course :)

Comment (nsoranzo) [reply lib/galaxy/web/framework/webapp.py]: Nice, thanks!

File: client/galaxy/scripts/mvc/workflow/workflow-view.js [hunk 2]
```diff
@@ -341,46 +353,48 @@ export default Backbone.View.extend({
...
-                    workflowNode.removeWorkflowOutput(output_name);
-                }
-                self.workflow.has_changes = true;
-            });
-            $("#workflow-output-area").show();
-        }
+        /* 
+         *
+         *  function edit_workflow_outputs() {
+         *      self.workflow.clear_active_node();
```
Comment (mvdbeek) [inline client/galaxy/scripts/mvc/workflow/workflow-view.js]: If we're not using this (doesn't look like we are) we can remove it, right ?

Comment (dannon) [reply client/galaxy/scripts/mvc/workflow/workflow-view.js]: Yeah, that was my intent.  I'll strip it, I meant to get back to that bit.

**Review by jmchilton — APPROVED** [review]

**Comment by mvdbeek** [comment]
Thanks @dannon!

**Pull request merged**